aws/hll/aws-s3-transfer-manager/src/download/object_meta.rs

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_sdk_s3::operation::{get_object::GetObjectOutput, head_object::HeadObjectOutput};
+
+// TODO(design): how many of these fields should we expose?
+// TODO(docs): Document fields
+
+/// Object metadata other than the body that can be set from either `GetObject` or `HeadObject`
+#[derive(Debug, Clone)]
+pub struct ObjectMetadata {
+    pub delete_marker: Option<bool>,
+    pub accept_ranges: Option<String>,
```

Comment:
Consider making this type a `Cow` so that it can avoid allocation in the case of the user providing a `&'static str`.

Reply:
This is all returned by either `GetObject` or `HeadObject` which is also of type `Option<String>` for the corresponding fields. Users don't set these directly.